src/WebJobs.Script.WebHost/Diagnostics/Sanitizer.cs

```diff
@@ -28,13 +29,29 @@ internal static string Sanitize(string input)
             }
 
             string t = input;
+            string inputWithPublicTokensHidden = input;
+
+            //Remove any known safe strings from the input before looking for Credentials
+            foreach (string publicToken in PublicTokens)
+            {
+                string hiddenString = string.Empty;
+                if (inputWithPublicTokensHidden.Contains(publicToken))
+                {
+                    foreach (char safechar in publicToken)
```

Comment:
Holy cow, I never knew this existed until I just looked it up: https://docs.microsoft.com/en-us/dotnet/api/system.string.-ctor?view=netframework-4.7.1#System_String__ctor_System_Char_System_Int32_

So you could do `string hiddenString = new string('#', publicToken.Length);` to remove this `foreach`